src/Type/Row.purs

```diff
@@ -32,3 +37,30 @@ instance rowLacks
      , RowLacking Entry key typ row )
   => RowLacks key row
 
+
+-- | A type-level list representation of a row
+foreign import kind RowList
+foreign import data Nil :: RowList
+foreign import data Cons :: Symbol -> Type -> RowList -> RowList
+
+-- | Extract the collection of entries in a closed row of types.
+-- | The list of entries is sorted by label and preserves duplicates.
+-- | The inverse of this operation is `ListToRow`.
+-- | Solved by the compiler.
+class RowToList (row :: # Type)
```

Comment:
I'd rather not, for now, since `prelude` is already getting quite big, but maybe we can discuss it for the next round of breaking changes.